frontend/departures.js

```diff
@@ -0,0 +1,59 @@
+let id = 0;
+
+function doFormSubmit() {
+    clearInterval(id);
+
+    // So that the every-30-second bus time updates continue using the same postcode on which they were originally called
+    // Save postcode as a variable, and pass to setInterval.
+    // clearInterval (above) will cancel the previous bus time update prescription when the form is next submitted
+    var postcode = getBusTimesReturnPostcode();
+    id = setInterval(getBusTimesReturnPostcode, 10000, postcode)
+}
+
+function getBusTimesReturnPostcode(postcode) {
+    var xhttp = new XMLHttpRequest();
+
+    if (!postcode) {
+        // The first time this function is called
+        // ie. not from setInterval, there will be no argument,
+        // so get the postcode from the form
+        var postcode = document.forms[0].elements["postcodeField"].value;
+    }
+
+    xhttp.open('GET', `/departureBoards?postcode=${postcode}`, true);
+    
+    xhttp.setRequestHeader('Content-Type', 'application/json');
+    
+    xhttp.onload = function() {
+        // Handle response here using e.g. xhttp.status, xhttp.response, xhttp.responseText
+        var resultsHTML = '<h2>Results</h2>';
+        
+        if (xhttp.status === 200) {

```

Comment:
Whilst you can't really do an early return here, I think inverting the condition would still help. It's nicer to read "if the status isn't 200, just log the response, otherwise (rest of function here)".